src/backend/app/main.py

```diff
@@ -43,9 +43,32 @@ async def health_check():
 
 @app.post("/generate_prediction/")
 async def generate_prediction(request: Request):
-    response = tools.generate_prediction(await request.json())
-    return JSONResponse(content=response)
+    eval_params=await request.json()
+    db = next(get_db())
+    predicted_hexagons=tools.get_predicted_hexagons(db, eval_params=eval_params)
+    if predicted_hexagons is None:
+        index_score_combined=tools.populate_prediction_database(eval_params,db)
```

Comment:
if `populate_prediction_database` returns a dictionary, this part can be simplified

```python
predicted_hexagons = [
    kex_index for kex_index, kex_score in  index_score_combined.items() 
    if  float(kex_score) >= eval_params["threshold"]
]
```

Reply:
It does not return a dictionary, it returns:
 index_score_combined = np.column_stack((hex_indexes, hex_scores)). 
but I accept other suggestions on making 
predicted_hexagons = [indexes[i] for i in range(len(indexes)) if scores[i]>= eval_params["threshold"]]
better!